sunpy/util/metadata.py

```diff
@@ -34,10 +34,12 @@ def __init__(self, *args):
             adict = args[0]
             if isinstance(adict, list) or isinstance(adict, tuple):
                 items = adict
-            elif isinstance(adict, dict):
```

Comment:
A better check might be to check for an instance of `collections.abc.MutableMapping`? It's generally better to avoid try except blocks if you can and I am like 90% sure that `fits.Header` is a Mapping.

Reply:
Good idea - I don't think being mutable matters here, so I just went for `Mapping`.